lib/ui/painting/image_descriptor.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_LIB_UI_PAINTING_IMAGE_DESCRIPTOR_H_
+#define FLUTTER_LIB_UI_PAINTING_IMAGE_DESCRIPTOR_H_
+
+#include <cstdint>
+#include <memory>
+
+#include "flutter/fml/macros.h"
+#include "flutter/lib/ui/dart_wrapper.h"
+#include "flutter/lib/ui/painting/immutable_buffer.h"
+#include "third_party/skia/include/codec/SkCodec.h"
+#include "third_party/skia/include/core/SkImageInfo.h"
+#include "third_party/tonic/dart_library_natives.h"
+
+namespace flutter {
+
+class ImageDescriptor : public RefCountedDartWrappable<ImageDescriptor> {
+ public:
+  ~ImageDescriptor() override = default;
+
+  static void initEncoded(Dart_NativeArguments args);
```

Comment:
This needs to be documented as it is not clear what the arguments and return types and error handling is supposed to look like.